packages/remirror__extension-code-block/src/code-block-extension.ts

```diff
@@ -293,29 +299,47 @@ export class CodeBlockExtension extends NodeExtension<CodeBlockOptions> {
    * Format the code block with the code formatting function passed as an
    * option.
    *
-   * The formatter runs asynchronously. This should not be immediately followed
-   * by additional commands. It is not yet possible to track completion of this
-   * command; it is best to use as a standalone response to a user action.
-   *
    * Code formatters (like prettier) add a lot to the bundle size and hence
    * it is up to you to provide a formatter which will be run on the entire
    * code block when this method is used.
    *
    * ```ts
-   * if (actions.formatCodeBlock.isActive()) {
-   *   actions.formatCodeBlockFactory();
-   *   // Or with a specific position
-   *   actions.formatCodeBlock({ pos: 100 }) // to format a separate code block
+   * if (actions.formatCodeBlock.enabled()) {
+   *   actions.formatCodeBlock();
+   *   // Or to format a separate code block via position
+   *   actions.formatCodeBlock({ pos: 100 });
    * }
    * ```
    */
   @command()
   formatCodeBlock(props?: Partial<PosProps>): CommandFunction {
-    return formatCodeBlockFactory({
-      type: this.type,
-      formatter: this.options.formatter,
-      defaultLanguage: this.options.defaultLanguage,
-    })(props);
+    return this.store
+      .createFormatCodeBlockCommand({
+        promise: () =>
+          formatCode({
+            type: this.type,
+            formatter: this.options.formatter,
+            defaultLanguage: this.options.defaultLanguage,
+            state: this.store.getState(),
+            ...props,
+          }),
```

Comment:
We should clean this bit up, using `props` here is confusing

`props` usually refers to `CommandFunctionProps`, but this is actually command _arguments_

```ts
formatCodeBlock({ pos }: Partial<PosProps> = {}): CommandFunction {
```

We can get the actual _command_ props, from the promise

```ts
      promise: (props) =>
          formatCode({
            type: this.type,
            formatter: this.options.formatter,
            defaultLanguage: this.options.defaultLanguage,
            pos, // The command argument,
            ...props, // CommandFunctionProps
          }),
```